<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Exploratory Data Analysis on [Researchers Name]’s data</title>
  <meta name="description" content="Exploratory Data Analysis on [Researchers Name]’s data">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Exploratory Data Analysis on [Researchers Name]’s data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Exploratory Data Analysis on [Researchers Name]’s data" />
  
  
  

<meta name="author" content="ProjectID [ProjectID] - [list researchers name] [bioinformaticians name]- Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI) - [email]@uni-mainz.de University Medical Center - Mainz (Germany)">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  


<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path=""><a href="#introduction"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path=""><a href="#qc-of-the-sequencing-samples"><i class="fa fa-check"></i><b>2</b> QC of the sequencing samples</a></li>
<li class="chapter" data-level="3" data-path=""><a href="#alignment-to-the-reference-genome"><i class="fa fa-check"></i><b>3</b> Alignment to the reference genome</a></li>
<li class="chapter" data-level="4" data-path=""><a href="#overview-and-creation-of-the-count-table"><i class="fa fa-check"></i><b>4</b> Overview and creation of the count table</a></li>
<li class="chapter" data-level="5" data-path=""><a href="#creation-of-the-design-matrix-and-sample-info"><i class="fa fa-check"></i><b>5</b> Creation of the design matrix and sample info</a></li>
<li class="chapter" data-level="6" data-path=""><a href="#samples-exploration-and-assessment"><i class="fa fa-check"></i><b>6</b> Samples exploration and assessment</a></li>
<li class="chapter" data-level="7" data-path=""><a href="#differential-expression-analysis"><i class="fa fa-check"></i><b>7</b> Differential Expression analysis</a><ul>
<li class="chapter" data-level="7.0.1" data-path=""><a href="#ko-del-vs-control"><i class="fa fa-check"></i><b>7.0.1</b> Ko Del Vs Control</a></li>
<li class="chapter" data-level="7.0.2" data-path=""><a href="#get-gene-names-and-annotations"><i class="fa fa-check"></i><b>7.0.2</b> Get Gene Names and Annotations</a></li>
<li class="chapter" data-level="7.0.3" data-path=""><a href="#result-of-ko-del-vs-control"><i class="fa fa-check"></i><b>7.0.3</b> Result of Ko Del Vs Control</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Exploratory Data Analysis on [Researchers Name]’s data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Exploratory Data Analysis on [Researchers Name]’s data</h1>
<h4 class="author"><em>ProjectID [ProjectID] - [list researchers name]</br> [bioinformaticians name]- Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI) - [email]<span class="citation">@uni-mainz.de</span></br> University Medical Center - Mainz (Germany)</em></h4>
<h4 class="date"><em>9 November 2018</em></h4>
</div>

<div id="introduction" class="section level1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">path_to_ref_genome &lt;-<span class="st"> &quot;/Users/franziska/Documents/Intro_rna_seq/data/human/GRChr38/annotation/Homo_sapiens.GRCh38.77.gtf&quot;</span>
is_paired_end &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;paired-end&quot;</span> =<span class="st"> </span><span class="ot">FALSE</span>, <span class="st">&quot;single-end&quot;</span> =<span class="st"> </span><span class="ot">TRUE</span>)
design_formula &lt;-<span class="st"> </span><span class="kw">as.formula</span>(<span class="st">&quot;~ condition&quot;</span>)
test_for &lt;-<span class="st"> &quot;ko_del_vs_control&quot;</span>
organ_dataset &lt;-<span class="st"> &quot;hsapiens_gene_ensembl&quot;</span></code></pre></div>
<p>RNA-seq samples were derived from [source_samples], obtaining a total of 5 different samples. More on the specific design is in the main sections.</p>
<!-- For each condition, biological triplicates were sequenced. -->
<p>For more information and an overview, see the figures and the tables in the main text.</p>
<p>All samples were sequenced with reads of [???] nucleotides, using a single-end library preparation (unstranded sequencing kit).</p>
<p>This report provides a detailed overview of the steps taken to perform a comprehensive exploratory data analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;DESeq2&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;ggplot2&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;ideal&quot;</span>)</code></pre></div>

</div>
<div id="qc-of-the-sequencing-samples" class="section level1">
<h1><span class="header-section-number">2</span> QC of the sequencing samples</h1>
<p>An overview of the sequenced samples is obtained with the FastQC tool (available at <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>). Inspecting the produced results, the quality is in line with good datasets.</p>

</div>
<div id="alignment-to-the-reference-genome" class="section level1">
<h1><span class="header-section-number">3</span> Alignment to the reference genome</h1>
<p>Short reads alignment was performed with the ENSEMBL Homo_sapiens.GRCh38.77 chosen as the reference genome. The corresponding annotation (ENSEMBL v76) was also retrieved from the ENSEMBL FTP website <a href="http://www.ensembl.org/info/data/ftp/index.html" class="uri">http://www.ensembl.org/info/data/ftp/index.html</a>.</p>
<p>The STAR aligner ([version number]) was used to perform the mapping to the reference genome.</p>

</div>
<div id="overview-and-creation-of-the-count-table" class="section level1">
<h1><span class="header-section-number">4</span> Overview and creation of the count table</h1>
<p>The available samples were reconstructed from the shared archive files. The raw <code>.fastq</code> files are available in compressed <code>.gz</code> format in the <code>_fastqFiles</code> folder.</p>
<p>The following samples are the relevant ones:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">list.files</span>(<span class="st">&quot;_alignedData/&quot;</span>,<span class="dt">pattern=</span><span class="st">&quot;(h_).*.bam$&quot;</span>)</code></pre></div>
<pre><code>## character(0)</code></pre>
<p>Following the read mapping step, we proceeded working with <code>.bam</code> files. We processed them with the <code>featureCounts</code> function of the <code>Rsubread</code> package, using the annotation file we also used for supporting the alignment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(bam_files &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;_alignedData/&quot;</span>,<span class="dt">pattern=</span><span class="st">&quot;(h_).*.bam$&quot;</span>,<span class="dt">full.names=</span><span class="ot">TRUE</span>))
fc_result &lt;-<span class="st"> </span>Rsubread<span class="op">::</span><span class="kw">featureCounts</span>(<span class="dt">files=</span>bam_files, 
                                     <span class="dt">annot.ext=</span>path_to_ref_genome,
                                     <span class="dt">isGTFAnnotationFile=</span><span class="ot">TRUE</span>,
                                     <span class="dt">nthreads=</span>parallel<span class="op">::</span><span class="kw">detectCores</span>()<span class="op">/</span><span class="dv">2</span>, <span class="dt">isPairedEnd=</span>is_paired_end[<span class="st">&quot;paired-end&quot;</span>])

<span class="kw">save</span>(fc_result,<span class="dt">file=</span><span class="st">&quot;results/fc_result.RData&quot;</span>)</code></pre></div>

</div>
<div id="creation-of-the-design-matrix-and-sample-info" class="section level1">
<h1><span class="header-section-number">5</span> Creation of the design matrix and sample info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samplesDesign &lt;-<span class="st"> </span><span class="kw">data.frame</span>(
  <span class="dt">fastqfile =</span> <span class="kw">list.files</span>(<span class="st">&quot;_fastqFiles&quot;</span>,<span class="dt">pattern =</span> <span class="st">&quot;.fastq.gz$&quot;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>),
  <span class="dt">bamfile =</span> <span class="kw">list.files</span>(<span class="st">&quot;_alignedData/&quot;</span>,<span class="dt">pattern =</span> <span class="st">&quot;.bam$&quot;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>),
  <span class="dt">condition =</span> <span class="kw">c</span>(<span class="st">&quot;control&quot;</span>, <span class="st">&quot;control&quot;</span>, <span class="st">&quot;ko_del&quot;</span>, <span class="st">&quot;ko_del&quot;</span>, <span class="st">&quot;ko_del&quot;</span>),
  <span class="dt">replNr =</span> <span class="kw">paste0</span>(<span class="st">&quot;R&quot;</span>,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)),
  <span class="dt">partner =</span> <span class="st">&quot;Alma&quot;</span>,
  <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>
)

samplesDesign<span class="op">$</span>sampleID &lt;-<span class="st"> </span><span class="kw">paste0</span>(samplesDesign<span class="op">$</span>condition,<span class="st">&quot;_&quot;</span>,samplesDesign<span class="op">$</span>replNr)
samplesDesign<span class="op">$</span>sampleName &lt;-<span class="st"> </span>samplesDesign<span class="op">$</span>sampleID
<span class="kw">write.csv</span>(samplesDesign, <span class="dt">file =</span> <span class="st">&quot;data/samplesDesign.csv&quot;</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;results/fc_result.RData&quot;</span>)
samplesDesign &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/samplesDesign.csv&quot;</span>)
samplesDesign<span class="op">$</span>bamfile &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;X_alignedData.&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="kw">colnames</span>(fc_result<span class="op">$</span>counts))

knitr<span class="op">::</span><span class="kw">kable</span>(samplesDesign)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">fastqfile</th>
<th align="left">bamfile</th>
<th align="left">condition</th>
<th align="left">replNr</th>
<th align="left">partner</th>
<th align="left">sampleID</th>
<th align="left">sampleName</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">_fastqFiles/Alma_1.fastq.gz</td>
<td align="left">Alma_1_Aligned.out.bam</td>
<td align="left">control</td>
<td align="left">R1</td>
<td align="left">Alma</td>
<td align="left">control_R1</td>
<td align="left">control_R1</td>
</tr>
<tr class="even">
<td align="left">_fastqFiles/Alma_2.fastq.gz</td>
<td align="left">Alma_2_Aligned.out.bam</td>
<td align="left">control</td>
<td align="left">R2</td>
<td align="left">Alma</td>
<td align="left">control_R2</td>
<td align="left">control_R2</td>
</tr>
<tr class="odd">
<td align="left">_fastqFiles/Alma_3.fastq.gz</td>
<td align="left">Alma_3_Aligned.out.bam</td>
<td align="left">ko_del</td>
<td align="left">R1</td>
<td align="left">Alma</td>
<td align="left">ko_del_R1</td>
<td align="left">ko_del_R1</td>
</tr>
<tr class="even">
<td align="left">_fastqFiles/Alma_4.fastq.gz</td>
<td align="left">Alma_4_Aligned.out.bam</td>
<td align="left">ko_del</td>
<td align="left">R2</td>
<td align="left">Alma</td>
<td align="left">ko_del_R2</td>
<td align="left">ko_del_R2</td>
</tr>
<tr class="odd">
<td align="left">_fastqFiles/Alma_5.fastq.gz</td>
<td align="left">Alma_5_Aligned.out.bam</td>
<td align="left">ko_del</td>
<td align="left">R3</td>
<td align="left">Alma</td>
<td align="left">ko_del_R3</td>
<td align="left">ko_del_R3</td>
</tr>
</tbody>
</table>

</div>
<div id="samples-exploration-and-assessment" class="section level1">
<h1><span class="header-section-number">6</span> Samples exploration and assessment</h1>
<p>We construct the dds objects which will contain all the required info for testing</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">counts &lt;-<span class="st"> </span>fc_result<span class="op">$</span>counts

dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromMatrix</span>(counts, samplesDesign, <span class="dt">design =</span> design_formula)
<span class="kw">colnames</span>(dds) &lt;-<span class="st"> </span>samplesDesign<span class="op">$</span>sampleID
rld_dds &lt;-<span class="st"> </span><span class="kw">rlogTransformation</span>(dds)</code></pre></div>
<p>Remove all the rows where no signal could be detected at all:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds &lt;-<span class="st"> </span>dds[<span class="kw">rowSums</span>(<span class="kw">counts</span>(dds)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>, ]
<span class="kw">nrow</span>(dds)</code></pre></div>
<pre><code>## [1] 18877</code></pre>
<p>The regularized log transformed values were used to perform sample clustering and principal component analysis (PCA).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pheatmap<span class="op">:::</span><span class="kw">pheatmap</span>(<span class="kw">as.matrix</span>(<span class="kw">dist</span>(<span class="kw">t</span>(<span class="kw">assay</span>(rld_dds)))))</code></pre></div>
<p><img src="_main_files/figure-html/sampleDist-1.png" width="672" /> [Comment on heatmap]</p>
<p>Create the PCA plots of all three datasets, again for comparison:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">top_genes &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">500</span>, <span class="dv">5000</span>, <span class="dv">20000</span>)
<span class="cf">for</span> (g <span class="cf">in</span> top_genes) {
  <span class="kw">print</span>(pcaExplorer<span class="op">:::</span><span class="kw">pcaplot</span>(rld_dds, <span class="dt">ntop=</span>g, <span class="dt">title =</span> <span class="kw">paste0</span>(<span class="st">&quot;top&quot;</span>, g, <span class="st">&quot; genes - global&quot;</span>), <span class="dt">ellipse =</span> F))
}</code></pre></div>
<p><img src="_main_files/figure-html/pcaSamples-1.png" width="672" /><img src="_main_files/figure-html/pcaSamples-2.png" width="672" /><img src="_main_files/figure-html/pcaSamples-3.png" width="672" /> [Comment on PCA plot]</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pcaExplorer<span class="op">::</span><span class="kw">pcascree</span>(<span class="kw">prcomp</span>(<span class="kw">t</span>(<span class="kw">assay</span>(rld_dds[,])),<span class="dt">center =</span> T))</code></pre></div>
<p><img src="_main_files/figure-html/Scree_plot-1.png" width="672" /></p>

</div>
<div id="differential-expression-analysis" class="section level1">
<h1><span class="header-section-number">7</span> Differential Expression analysis</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">FDR &lt;-<span class="st"> </span><span class="fl">0.05</span></code></pre></div>
<p>We set the False Discovery Rate to 5%, and perform the following steps, we obtain the full set of Differentially Expressed genes, i.e. the ones that (regardless of the lfc) show robust significant differences in the expression among conditions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds)</code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div id="ko-del-vs-control" class="section level3">
<h3><span class="header-section-number">7.0.1</span> Ko Del Vs Control</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">resultsNames</span>(dds)</code></pre></div>
<pre><code>## [1] &quot;Intercept&quot;                   &quot;condition_ko_del_vs_control&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_condition &lt;-<span class="st"> </span><span class="kw">results</span>(dds, <span class="dt">name =</span> <span class="kw">paste0</span>(<span class="st">&quot;condition_&quot;</span>, test_for),
                         <span class="dt">alpha =</span> FDR)
res_condition &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(dds, <span class="dt">coef =</span> <span class="kw">paste0</span>(<span class="st">&quot;condition_&quot;</span>, test_for),
                           <span class="dt">type =</span> <span class="st">&quot;apeglm&quot;</span>, <span class="dt">res =</span> res_condition) </code></pre></div>
<pre><code>## using &#39;apeglm&#39; for LFC shrinkage. If used in published research, please cite:
##     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
##     sequence count data: removing the noise and preserving large differences.
##     bioRxiv. https://doi.org/10.1101/303255</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(res_condition)</code></pre></div>
<pre><code>## 
## out of 18877 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 669, 3.5%
## LFC &lt; 0 (down)     : 342, 1.8%
## outliers [1]       : 3, 0.016%
## low counts [2]     : 6222, 33%
## (mean count &lt; 6)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_ma</span>(res_condition, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">2</span>,<span class="dv">2</span>), <span class="dt">title =</span> <span class="kw">paste</span>(<span class="st">&quot;Result of&quot;</span>,                                         stringr<span class="op">::</span><span class="kw">str_to_title</span>(
  stringr<span class="op">::</span><span class="kw">str_replace_all</span>(test_for, <span class="dt">pattern =</span> <span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>))), <span class="dt">FDR =</span> FDR)</code></pre></div>
<p><img src="_main_files/figure-html/retrieve_results-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tbl_res_condition &lt;-<span class="st"> </span><span class="kw">deseqresult2tbl</span>(res_condition)
tbl_res_alma_ko_vs_ctrl<span class="op">$</span>geneSymbol &lt;-<span class="st"> </span>anno_df<span class="op">$</span>gene_name[<span class="kw">match</span>(tbl_res_alma_ko_vs_ctrl<span class="op">$</span>id, anno_df<span class="op">$</span>gene_id)]
tbl_res_alma_ko_vs_ctrl<span class="op">$</span>description &lt;-<span class="st"> </span>anns<span class="op">$</span>description[<span class="kw">match</span>(tbl_res_alma_ko_vs_ctrl<span class="op">$</span>id, anns<span class="op">$</span>ensembl_gene_id)]




mart &lt;-<span class="st"> </span>biomaRt<span class="op">:::</span><span class="kw">useMart</span>(<span class="dt">biomart=</span><span class="st">&quot;ENSEMBL_MART_ENSEMBL&quot;</span>, <span class="dt">host=</span><span class="st">&quot;www.ensembl.org&quot;</span>, <span class="dt">dataset=</span>organ_dataset)
anns &lt;-<span class="st"> </span><span class="kw">getBM</span>( <span class="dt">attributes =</span> <span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>, <span class="st">&quot;external_gene_name&quot;</span>, <span class="st">&quot;description&quot;</span>), <span class="dt">filters =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>, <span class="dt">values =</span> <span class="kw">rownames</span>(dds_marie), <span class="dt">mart =</span> mart)
anns &lt;-<span class="st"> </span>anns[<span class="kw">match</span>(<span class="kw">rownames</span>(dds_marie), anns[, <span class="dv">1</span>]), ]
anno_df &lt;-<span class="st"> </span>pcaExplorer<span class="op">::</span><span class="kw">get_annotation</span>(dds_marie,<span class="st">&quot;hsapiens_gene_ensembl&quot;</span>,<span class="st">&quot;ensembl_gene_id&quot;</span>)
tbl_DEres_marie_inside_vs_outside &lt;-<span class="st"> </span><span class="kw">deseqresult2DEgenes</span>(res_marie_inside_vs_outside,<span class="dt">FDR =</span> FDR)
tbl_DEres_marie_inside_vs_outside<span class="op">$</span>geneSymbol &lt;-<span class="st"> </span>anno_df<span class="op">$</span>gene_name[<span class="kw">match</span>(tbl_DEres_marie_inside_vs_outside<span class="op">$</span>id, anno_df<span class="op">$</span>gene_id)]
tbl_DEres_marie_inside_vs_outside<span class="op">$</span>description &lt;-<span class="st"> </span>anns<span class="op">$</span>description[<span class="kw">match</span>(tbl_DEres_marie_inside_vs_outside<span class="op">$</span>id, anns<span class="op">$</span>ensembl_gene_id)]

etbl_DEres_marie_inside_vs_outside &lt;-<span class="st"> </span>tbl_DEres_marie_inside_vs_outside
etbl_DEres_marie_inside_vs_outside<span class="op">$</span>id &lt;-<span class="st"> </span>ideal<span class="op">:::</span><span class="kw">createLinkENS</span>(etbl_DEres_marie_inside_vs_outside<span class="op">$</span>id, <span class="dt">species =</span> <span class="st">&quot;Homo_sapiens&quot;</span>)
etbl_DEres_marie_inside_vs_outside<span class="op">$</span>geneSymbol &lt;-<span class="st"> </span>ideal<span class="op">:::</span><span class="kw">createLinkGeneSymbol</span>(etbl_DEres_marie_inside_vs_outside<span class="op">$</span>geneSymbol)
mybuttons &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;copy&#39;</span>, <span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>, <span class="st">&#39;pdf&#39;</span>, <span class="st">&#39;print&#39;</span>)
<span class="kw">datatable</span>(etbl_DEres_marie_inside_vs_outside,<span class="dt">caption =</span> <span class="st">&quot;petridish_tumor_vs_primtum, DE genes&quot;</span>,<span class="dt">escape=</span>F, <span class="dt">extensions =</span> <span class="st">&#39;Buttons&#39;</span>, <span class="dt">options =</span> <span class="kw">list</span>(<span class="dt">pageLength =</span> <span class="dv">50</span>,<span class="dt">dom =</span> <span class="st">&#39;Bfrtip&#39;</span>,<span class="dt">buttons =</span> mybuttons))</code></pre></div>

</div>
<div id="get-gene-names-and-annotations" class="section level3">
<h3><span class="header-section-number">7.0.2</span> Get Gene Names and Annotations</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bla &lt;-<span class="st"> </span><span class="dv">0</span></code></pre></div>

</div>
<div id="result-of-ko-del-vs-control" class="section level3">
<h3><span class="header-section-number">7.0.3</span> Result of Ko Del Vs Control</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tbl_res_condition &lt;-<span class="st"> </span><span class="kw">deseqresult2tbl</span>(res_condition)
tbl_res_alma_ko_vs_ctrl<span class="op">$</span>geneSymbol &lt;-<span class="st"> </span>anno_df<span class="op">$</span>gene_name[<span class="kw">match</span>(tbl_res_alma_ko_vs_ctrl<span class="op">$</span>id, anno_df<span class="op">$</span>gene_id)]
tbl_res_alma_ko_vs_ctrl<span class="op">$</span>description &lt;-<span class="st"> </span>anns<span class="op">$</span>description[<span class="kw">match</span>(tbl_res_alma_ko_vs_ctrl<span class="op">$</span>id, anns<span class="op">$</span>ensembl_gene_id)]</code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>


    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
