# Introduction

```{r setup_variables}
path_to_ref_genome <- "/Users/franziska/Documents/Intro_rna_seq/data/human/GRChr38/annotation/Homo_sapiens.GRCh38.77.gtf"
is_paired_end <- c("paired-end" = FALSE, "single-end" = TRUE)
design_formula <- as.formula("~ condition")
test_for <- "ko_del_vs_control"
organ_dataset <- "hsapiens_gene_ensembl"
```

```{r setup_variables_hidden, include=FALSE}
samplesDesign <- read.csv("data/samplesDesign.csv")
```

RNA-seq samples were derived from [source_samples], obtaining a total of `r nrow(samplesDesign)` different samples. More on the specific design is in the main sections.

<!-- For each condition, biological triplicates were sequenced. -->
For more information and an overview, see the figures and the tables in the main text.

All samples were sequenced with reads of [???] nucleotides, using a `r names(is_paired_end[is_paired_end==T])` library preparation (unstranded sequencing kit). 

This report provides a detailed overview of the steps taken to perform a comprehensive exploratory data analysis.

```{r loadLibraries, results='hide'}
library("DESeq2")
library("ggplot2")
library("ideal")
```

